### PR TITLE
e2e_dra: stop using deprecated framework.ExpectEqual

### PR DESCRIPTION
**Pull request by carlory**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

e2e_dra: stop using deprecated framework.ExpectEqual

#### Which issue(s) this PR fixes:

Ref https://github.com/kubernetes/kubernetes/pull/115961

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: test/e2e/dra/dra.go [hunk 1]
```diff
@@ -647,7 +647,7 @@ var _ = ginkgo.Describe("[sig-node] DRA [Feature:DynamicResourceAllocation]", fu
 					framework.Fail("first pod should be running on node, was not scheduled")
 				}
 				framework.ExpectNotEqual(pod1.Spec.NodeName, node, "first pod should run on different node than second one")
```
Comment (TommyStarK) [inline test/e2e/dra/dra.go:649]: Is it possible to update this line as well ?

```suggestion
gomega.Expect(pod1.Spec.NodeName).ToNot(gomega.Equal(node), "first pod should run on different node than second one")
```

**Comment by carlory** [comment]
@bart0sh @TommyStarK please review again.

**Review by swatisehgal — COMMENTED** [review]
/lgtm

**Pull request merged**